src/System.Text.Json/tests/JsonNodeTransformationsTests.cs

```diff
@@ -0,0 +1,187 @@
+        }
+
+        [Fact]
+        public static void TestJsonDocumentToJsonNode()
+        {
+            JsonDocument jsonDocument = JsonDocument.Parse(jsonSampleString);
+            JsonNode node = JsonNode.DeepCopy(jsonDocument);
+            CheckNodeFromSampleString(node);
+        }
+
+        [Fact]
+        public static void TestParseStringToJsonNode()
+        {
+            JsonNode node = JsonNode.Parse(jsonSampleString);
+            CheckNodeFromSampleString(node);
+        }
+
+        [Fact]
+        public static void TestParseBytesToJsonNode()
+        {
+            JsonNode node = JsonNode.Parse(new ReadOnlySequence<byte>(Encoding.UTF8.GetBytes(jsonSampleString).AsMemory())); 
+            CheckNodeFromSampleString(node);
+        }
+
+        [Fact]
+        public static void TestParseByteMemoryToJsonNode()
+        {
+            JsonNode node = JsonNode.Parse(Encoding.UTF8.GetBytes(jsonSampleString).AsMemory());
+            CheckNodeFromSampleString(node);
+        }
+
+        [Fact]
+        public static void TestParseCharMemoryToJsonNode()
+        {
+            JsonNode node = JsonNode.Parse(jsonSampleString.AsMemory());
+            CheckNodeFromSampleString(node);
+        }
+
+        [Fact]
+        public static void TestParseStreamToJsonNode()
+        {
+            JsonNode node = JsonNode.Parse(new MemoryStream(Encoding.UTF8.GetBytes(jsonSampleString)));
+            CheckNodeFromSampleString(node);
+        }
+
+        [Fact]
+        public static void TestCloneJsonArray()
+        {
+            var jsonArray = new JsonArray { "value1", "value2" };
+            var jsonArrayCopy = JsonNode.DeepCopy(jsonArray) as JsonArray;
+            Assert.Equal(2, jsonArrayCopy.Count);
+            jsonArray.Add("value3");
+            Assert.Equal(2, jsonArrayCopy.Count);
+        }
+
+        [Fact]
+        public static void TestCloneJsonNode()
+        {
+            var jsonObject = new JsonObject
+            {
+                { "text", "property value" },
+                { "boolean", true },
+                { "number", 15 },
+                { "array", new JsonString[] { "value1", "value2"} }
+            };
+
+            var jsonObjectCopy = (JsonObject)JsonNode.DeepCopy(jsonObject);
+
+            jsonObject["text"] = (JsonString)"something different";
+            Assert.Equal("property value", (JsonString)jsonObjectCopy["text"]);
+
+            ((JsonBoolean)jsonObject["boolean"]).Value = false;
+            Assert.True(((JsonBoolean)jsonObjectCopy["boolean"]).Value);
+
+            Assert.Equal(2, jsonObjectCopy.GetJsonArrayProperty("array").Count);
+            jsonObject.GetJsonArrayProperty("array").Add("value3");
+            Assert.Equal(2, jsonObjectCopy.GetJsonArrayProperty("array").Count);
+
+            jsonObject.Add("new one", 123);
+            Assert.Equal(4, jsonObjectCopy.PropertyNames.Count);
+        }
+
+        [Fact]
+        public static void TestAsJsonElement()
```

Comment:
Might be worth splitting this into two different classes / purposes: JsonNodeToJsonElementTest and JsonNodeDeepCopyTests